pypeit/reduce.py

```diff
@@ -443,14 +436,36 @@ def run_extraction(self, global_sky, sobjs_obj, skymask, ra=None, dec=None, obst
         for i, sobj in enumerate(sobjs_obj):
             if sobj.SLITID in list(self.slits.spat_id[self.reduce_bpm]):
                 remove_idx.append(i)
+        # remove
         sobjs_obj.remove_sobj(remove_idx)
 
+        # # Create skymask for maskdef_extracted objects
+        if self.par['reduce']['slitmask']['extract_missing_objs']:
+            gdslits = np.where(np.invert(self.reduce_bpm))[0]
+            # Loop on slits
+            for slit_idx in gdslits:
+                slit_spat = self.slits.spat_id[slit_idx]
+                slit_objs = sobjs_obj[sobjs_obj.SLITID == slit_spat]
+                thismask = self.slitmask == slit_spat
+                # sobj index of maskdef_extract
+                maskdef_extract = np.where(slit_objs.MASKDEF_EXTRACT == True)[0] if slit_objs.nobj > 0 else np.array([])
+                if maskdef_extract.size > 0:
+                    skymask_fwhm = extract.create_skymask_fwhm(slit_objs[maskdef_extract], skymask,
+                            box_pix=self.par['reduce']['extraction']['boxcar_radius']/self.get_platescale(None) \
+                                    if self.par['reduce']['skysub']['mask_by_boxcar'] else None)
+                    skymask[thismask] = skymask_fwhm[thismask]
         self.sobjs_obj = sobjs_obj
         self.skymask = skymask
         self.nobj = len(sobjs_obj)
 
         # Do we have any positive objects to proceed with?
         if self.nobj > 0:
```

Comment:
do we want to insist `self.nobj>0` ?
  I think we want a 2nd pass either way as we will get a better sky subtracted slit.

Reply:
I decided to re-add `self.nobj>0` because I thought that if there are no objects found at all (including `maskdef_extract` objects), the global sky should not be different than the `initial_sky` and therefore it would save some time not running a second global sky subtraction. But I'm OK to remove  `self.nobj>0` if you think it is better.